File: php/property.php
<?php
class Rectangle {

	private $width = 0;
	private $height = 0;

	public function info() {
		echo "Width: $this->width, Height: $this->height \n";
	}

	function __set($property, $value) {
		if ($property == 'w') {
			if ($value < $this->width) {
				$this->height = $value;
				echo "yay";
			}
			else {
				$this->height = $this->width;
				$this->width = $value;
			}
		}
	}

	function __get($property) {
		switch ($property) {
			case 'w': return $this->width + 2;
			case 'h': return $this->height + 1;
		}
	}

}

$r = new Rectangle();
$r->w = 50;
$r->w = 20;
$r->info();
echo "w: $r->w, h: $r->h\n";

?>

File: python/property.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-

class Rectangle(object):

	def __init__(self):
		self.width = 0
		self.height = 0

	def info(self):
		print("Width: " + str(self.width) + ", Height: " + str(self.height))

	@property
	def w(self):
		return self.width + 2

	@property
	def h(self):
		return self.height + 1

	@w.setter
	def w(self, value):
		if value < self.width :
			self.height = value
		else :
			self.height = self.width
			self.width = value

r = Rectangle();
r.w = 50;
r.w = 20;
r.info();
print("w: " + str(r.w) + ", h: " + str(r.h));
